src/AspNet.Security.OAuth.Reddit/RedditAuthenticationHandler.cs

```diff
@@ -32,6 +32,10 @@ public RedditAuthenticationHandler([NotNull] HttpClient client)
             request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
             request.Headers.Authorization = new AuthenticationHeaderValue("bearer", tokens.AccessToken);
 
+            if (!string.IsNullOrWhiteSpace(Options.UserAgent)) {
+                request.Headers.Add("User-Agent", Options.UserAgent);
```

Comment:
Because the other header is set with `.DefaultHeaders` it is overridden correctly when it's expressly supplied with an actual value in the request.

Unfortunately I can't get httpclient tracing to behave properly for the request, but manually stepping through the request and response indicate that it is working properly.